pkg/api/admin/openshiftcluster_convert.go

```diff
@@ -342,3 +351,12 @@ func (c openShiftClusterConverter) ToInternal(_oc interface{}, out *api.OpenShif
 	// with filling the out.Properties.RegistryProfiles[i].Password as default is "" which erases the original value.
 	// Workaround would be filling the password when receiving request, but it is array and the logic would be to complex.
 }
+
+// ExternalNoReadOnly removes all read-only fields from the external representation.
+func (c openShiftClusterConverter) ExternalNoReadOnly(_oc interface{}) {
```

Comment:
This can be done using the reflection package to make us of the swagger field. Same is done for RP config here https://github.com/mbarnes/ARO-RP/blob/452d153e8b413ad28efadb9b7e5b989d89c59aac/pkg/deploy/config.go#L166-L188. 

No need to block on this, but I would like to see done as a follow up, please. This has the potential to be forgotten over time when setting field in config structure and forgetting it here.

Reply:
I think it's appropriate to do as a follow up.  I looked into doing this with `reflect` but it's going to require recursion since these tags can be several levels deep in nested struct types.

What might be best is see if I can factor out the recursive type walking in [immutable.go](https://github.com/Azure/ARO-RP/blob/master/pkg/api/util/immutable/immutable.go), apply the [visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern) through a callback parameter, and make some kind of reusable type inspection utility.